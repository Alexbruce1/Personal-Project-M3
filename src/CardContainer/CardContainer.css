.outer-container {
  background: linear-gradient(to bottom, transparent, #fcfcfc 1000px, #fcfcfc);
  /* background-color: transparent; */
  position: absolute;
  top: 15rem;
  margin: 5rem 0 0 0;
  width: 100%;
}

.type-button-container {
  display: inline-flex;
  justify-content: space-between;
  margin: 1rem auto;
  width: calc(100% - 2rem);
}

.recommendations-slider {
  /* background-color: rgba(242, 242, 242, .95); */
  border-radius: 8px;
  box-shadow: 0 0 15px rgba(0, 0, 0, .4);
  height: 10rem;
  margin: 8px auto;
  width: calc(100% - 2rem);
}

.card-container {
  /* background-color: rgba(242, 242, 242, .5); */
  border-radius: 8px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  column-gap: 1rem;
  padding: 5px 1rem;
  width: calc(100% - 2rem);
  /* overflow: scroll; */
}
