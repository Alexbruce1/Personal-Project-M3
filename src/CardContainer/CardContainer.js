import React from 'react';
import Card from '../Card/Card';
import dinosaur from '../images/dinosaur.png';
import './CardContainer.css';


const CardContainer = (props) => {
  const postData = props.media.map((item, i) => {
    return <Card
      key={i}
      name={item.name}
      type={item.type}
      teaser={item.teaser}
      wiki={item.wiki}
      yt={item.yt}
      yId={item.yId} />
  })


  return(
    <div className='outer-container'>
      <div className='card-container'>
        { props.media.length > 1 ?
          postData :
          <div className='no-responses'>
            <h2>Sorry, there were no matches to your search</h2>
            <img alt='' src={dinosaur} />
            <h2>Try another search</h2>
          </div>
        }
      </div>
    </div>
  )
}

export default CardContainer;