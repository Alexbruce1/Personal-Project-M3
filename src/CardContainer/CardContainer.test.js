import CardContainer from './CardContainer';
import React from 'react';
import { shallow } from 'enzyme';


describe('CardContainer', () => {
  let wrapper;
  let data;

  beforeEach(() => {
    data = [{}, {}];
    wrapper = shallow(<CardContainer media={data} />);
  });

  it('should render like the snapshot with data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render like the snapshot without data', () => {
    wrapper = shallow(<CardContainer media={[]} />);
    
    expect(wrapper).toMatchSnapshot();
  })
})