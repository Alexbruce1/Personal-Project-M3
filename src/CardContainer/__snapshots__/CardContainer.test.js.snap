// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render like the snapshot with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    media={
      Array [
        Object {},
        Object {},
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-container"
      >
        <Card />
        <Card />
      </div>,
      "className": "outer-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card />,
          <Card />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "name": undefined,
            "teaser": undefined,
            "type": undefined,
            "wiki": undefined,
            "yId": undefined,
            "yt": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "name": undefined,
            "teaser": undefined,
            "type": undefined,
            "wiki": undefined,
            "yId": undefined,
            "yt": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-container"
        >
          <Card />
          <Card />
        </div>,
        "className": "outer-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card />,
            <Card />,
          ],
          "className": "card-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "name": undefined,
              "teaser": undefined,
              "type": undefined,
              "wiki": undefined,
              "yId": undefined,
              "yt": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "name": undefined,
              "teaser": undefined,
              "type": undefined,
              "wiki": undefined,
              "yId": undefined,
              "yt": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`CardContainer should render like the snapshot without data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    media={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-container"
      >
        <div
          className="no-responses"
        >
          <h2>
            Sorry, there were no matches to your search
          </h2>
          <img
            alt=""
            src="dinosaur.png"
          />
          <h2>
            Try another search
          </h2>
        </div>
      </div>,
      "className": "outer-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="no-responses"
        >
          <h2>
            Sorry, there were no matches to your search
          </h2>
          <img
            alt=""
            src="dinosaur.png"
          />
          <h2>
            Try another search
          </h2>
        </div>,
        "className": "card-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Sorry, there were no matches to your search
            </h2>,
            <img
              alt=""
              src="dinosaur.png"
            />,
            <h2>
              Try another search
            </h2>,
          ],
          "className": "no-responses",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sorry, there were no matches to your search",
            },
            "ref": null,
            "rendered": "Sorry, there were no matches to your search",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "src": "dinosaur.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Try another search",
            },
            "ref": null,
            "rendered": "Try another search",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-container"
        >
          <div
            className="no-responses"
          >
            <h2>
              Sorry, there were no matches to your search
            </h2>
            <img
              alt=""
              src="dinosaur.png"
            />
            <h2>
              Try another search
            </h2>
          </div>
        </div>,
        "className": "outer-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="no-responses"
          >
            <h2>
              Sorry, there were no matches to your search
            </h2>
            <img
              alt=""
              src="dinosaur.png"
            />
            <h2>
              Try another search
            </h2>
          </div>,
          "className": "card-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Sorry, there were no matches to your search
              </h2>,
              <img
                alt=""
                src="dinosaur.png"
              />,
              <h2>
                Try another search
              </h2>,
            ],
            "className": "no-responses",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sorry, there were no matches to your search",
              },
              "ref": null,
              "rendered": "Sorry, there were no matches to your search",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "src": "dinosaur.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Try another search",
              },
              "ref": null,
              "rendered": "Try another search",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
